File: src/main/java/com/examportal/dto/UserSignInDTO.java
package com.examportal.dto;

import jakarta.validation.constraints.NotBlank;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@ToString
public class UserSignInDTO {
    
    @NotBlank(message = "Email or Username must be provided")
    private String emailOrUsername;
    
    @NotBlank(message = "Password must be provided")
    private String password;
}

File: Dotnet3/ExamPortal/DTOs/UserSignInDTO.cs
using System.ComponentModel.DataAnnotations;

namespace ExamPortal.DTOs
{
    public class UserSignInDTO
    {
        [Required(ErrorMessage = "Email or Username must be provided")]
        public string EmailOrUsername { get; set; } = string.Empty;
        
        [Required(ErrorMessage = "Password must be provided")]
        public string Password { get; set; } = string.Empty;
    }
}
